src/ripple/rpc/tests/KeyGeneration.test.cpp

```diff
@@ -0,0 +1,174 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2015 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/basics/StringUtilities.h>
+#include <ripple/basics/TestSuite.h>
+#include <ripple/rpc/handlers/WalletPropose.h>
+#include <ripple/rpc/impl/KeypairForSignature.h>
+
+namespace ripple {
+
+namespace RPC {
+
+struct key_strings
+{
+    char const* account_id;
+    char const* master_key;
+    char const* master_seed;
+    char const* master_seed_hex;
+    char const* public_key;
+    char const* public_key_hex;
+    char const* secret_key_hex;
+};
+
+namespace common {
+static char const* passphrase = "REINDEER FLOTILLA";
+static char const* master_key = "SCAT BERN ISLE FOR ROIL BUS SOAK AQUA FREE FOR DRAM BRIG";
+static char const* master_seed = "snMwVWs2hZzfDUF3p2tHZ3EgmyhFs";
+static char const* master_seed_hex = "BE6A670A19B209E112146D0A7ED2AAD7";
+}
+
+static key_strings const secp256k1_strings =
+{
+    "r4Vtj2jrfmTVZGfSP3gH9hQPMqFPQFin8f",
+    common::master_key,
+    common::master_seed,
+    common::master_seed_hex,
+    "aBQxK2YFNqzmAaXNczYcjqDjfiKkLsJUizsr1UBf44RCF8FHdrmX",
+    "038AAE247B2344B1837FBED8F57389C8C11774510A3F7D784F2A09F0CB6843236C",
+    "1949ECD889EA71324BC7A30C8E81F4E93CB73EE19D59E9082111E78CC3DDABC2",
+};
+
+class WalletPropose_test : public ripple::TestSuite
+{
+public:
+    void testRandomWallet()
+    {
+        Json::Value params;
+        Json::Value result = WalletPropose (params);
+
+        expect (! contains_error (result));
+        expect (result.isMember ("account_id"));
+        expect (result.isMember ("master_key"));
+        expect (result.isMember ("master_seed"));
+        expect (result.isMember ("master_seed_hex"));
+        expect (result.isMember ("public_key"));
+        expect (result.isMember ("public_key_hex"));
+
+        std::string seed = result["master_seed"].asString();
+
+        result = WalletPropose (params);
+
+        // We asked for two random seeds, so they shouldn't match.
```

Comment:
Thanks for the complete-seeming tests!

To be honest, my understanding isn't quite great enough to propose another new test, even a ridiculous one that wouldn't be worth doing... if you did write another test, what might it be?  (I won't ask you to do it unless it's super interesting, I'm just curious as to what assumptions are being made...)